src/Conversion/ONNXToKrnl/Math/QLinearMatMul.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===--------- QLinearMatMul.cpp - Lowering QLinearMatMul Op --------------===//
+//
+// Copyright 2019-2025 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX QLinearMatMul Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/Krnl/DialectBuilder.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
+#include "src/Dialect/Mlir/DialectBuilder.hpp"
+#include "src/Dialect/Mlir/IndexExpr.hpp"
+#include "src/Dialect/ONNX/ONNXOps/ShapeHelper.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+struct ONNXQLinearMatMulOpLowering
+    : public OpConversionPattern<ONNXQLinearMatMulOp> {
+public:
+  ONNXQLinearMatMulOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : OpConversionPattern(typeConverter, ctx) {}
+
+  LogicalResult matchAndRewrite(ONNXQLinearMatMulOp qlmmOp,
+      ONNXQLinearMatMulOpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const final {
+    using LocalDialectBuilder =
+        MultiDialectBuilder<IndexExprBuilderForKrnl, OnnxBuilder>;
+    Operation *op = qlmmOp.getOperation();
+    Location loc = ONNXLoc<ONNXQLinearMatMulOp>(op);
+    LocalDialectBuilder create(rewriter, loc);
+
+    ValueRange operands = adaptor.getOperands();
+    Value A = adaptor.getA();
+    Value aScale = adaptor.getAScale();
+    Value aZeroPoint = adaptor.getAZeroPoint();
+    Value B = adaptor.getB();
+    Value bScale = adaptor.getBScale();
+    Value bZeroPoint = adaptor.getBZeroPoint();
+    Value yScale = adaptor.getYScale();
+    Value yZeroPoint = adaptor.getYZeroPoint();
+
+    // Only support integer8 and float32 now.
+    if (!getElementType(A.getType()).isInteger(8))
```

Comment:
integer8 for A and B, and flaot32 for zeroPoint?

Reply:
Yes, made it clear now. Thanks!